server/controllers/personal/personalController.ts

```diff
@@ -423,6 +424,109 @@ export default class PersonalController {
     }
   }
 
+  editRadioFields(formTitle: string, fieldData: RadioFieldData, options: SelectOption[]) {
+    return async (req: Request, res: Response, next: NextFunction) => {
+      const { pageTitle, hintText, auditPage } = fieldData
+      const { prisonerNumber } = req.params
+      const { prisonerData } = req.middleware
+      const { firstName, lastName, cellLocation } = prisonerData
+      const errors = req.flash('errors')
+
+      const prisonerBannerName = formatName(firstName, null, lastName, { style: NameFormatStyle.lastCommaFirst })
+
+      await this.auditService.sendPageView({
+        user: res.locals.user,
+        prisonerNumber: prisonerData.prisonerNumber,
+        prisonId: prisonerData.prisonId,
+        correlationId: req.id,
+        page: auditPage,
+      })
+
+      res.render('pages/edit/radioField', {
+        pageTitle: `${pageTitle} - Prisoner personal details`,
+        formTitle,
+        prisonerNumber,
+        breadcrumbPrisonerName: prisonerBannerName,
+        errors,
+        hintText,
+        options,
+        miniBannerData: {
+          prisonerName: prisonerBannerName,
+          prisonerNumber,
+          cellLocation: formatLocation(cellLocation),
+        },
+      })
+    }
+  }
+
+  smokerOrVaper() {
+    const fieldData: RadioFieldData = {
+      auditPage: Page.EditSmokerOrVaper,
+      fieldName: 'smokerOrVaper',
+      pageTitle: 'Smoker or vaper',
+      url: 'smoker-or-vaper',
+    }
+
+    return {
+      edit: async (req: Request, res: Response, next: NextFunction) => {
+        const { inmateDetail, prisonerData } = req.middleware
+        const { firstName, lastName } = prisonerData
+        const requestBodyFlash = requestBodyFromFlash<{ radioField: string }>(req)
+        const fieldValue =
+          requestBodyFlash?.radioField ||
+          getProfileInformationValue(ProfileInformationType.SmokerOrVaper, inmateDetail.profileInformation)
+
+        // Placeholder for now
+        const options: SelectOption[] = [
+          {
+            text: 'Yes',
+            value: 'Yes',
+            selected: fieldValue === 'Yes',
```

Comment:
These are radios, so needs to be `checked` to set the radio to "on"
Realised that I've also been using `SelectOption[]` for radios as the govuk design system has the `items` property for selects and radios as the same structure, so generally it just works.
Probably would be good to define a new interface for `RadioOption` (or something like that) so we can put `checked` on there rather than on `SelectOption` which already has `selected`